components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIFileUtil.java

```diff
@@ -0,0 +1,229 @@
+        }
+    }
+
+    /**
+     * Creates a zip archive from the given {@link InputStream} inputStream
+     *
+     * @param inputStream {@link InputStream} instance
+     * @param archivePath path to create the zip archive
+     * @throws APIManagementException if an error occurs while creating the archive
+     */
+    public static void createArchiveFromInputStream(InputStream inputStream, String archivePath)
+            throws APIManagementException {
+        try (FileOutputStream outFileStream = new FileOutputStream(new File(archivePath))) {
+            IOUtils.copy(inputStream, outFileStream);
+        } catch (IOException e) {
+            String errorMsg = "Error in Creating archive from data.";
+            log.error(errorMsg, e);
+            throw new APIManagementException(errorMsg, e);
+        }
+    }
+
+    /**
+     * Extracts a a given zip archive
+     *
+     * @param archiveFilePath path of the zip archive
+     * @param destination     extract location
+     * @return name of the extracted zip archive
+     * @throws APIManagementException if an error occurs while extracting the archive
+     */
+    public static String extractArchive(String archiveFilePath, String destination)
+            throws APIManagementException {
+        String archiveName = null;
+
+        try (ZipFile zip = new ZipFile(new File(archiveFilePath))) {
+            Enumeration zipFileEntries = zip.entries();
+            int index = 0;
+
+            // Process each entry
+            while (zipFileEntries.hasMoreElements()) {
+
+                // grab a zip file entry
+                ZipEntry entry = (ZipEntry) zipFileEntries.nextElement();
+                String currentEntry = entry.getName();
+
+                //This index variable is used to get the extracted folder name; that is root directory
+                if (index == 0 && currentEntry.indexOf('/') != -1) {
+                    archiveName = currentEntry.substring(0, currentEntry.indexOf('/'));
+                    --index;
+                }
+
+                File destinationFile = new File(destination, currentEntry);
+                File destinationParent = destinationFile.getParentFile();
+                String canonicalizedDestinationFilePath = destinationFile.getCanonicalPath();
+
+                if (!canonicalizedDestinationFilePath.startsWith(new File(destination).getCanonicalPath())) {
+                    String errorMessage = "Attempt to upload invalid zip archive with file at " + currentEntry +
+                            ". File path is outside target directory";
+                    log.error(errorMessage);
+                    throw new APIManagementException(errorMessage);
+                }
+
+                // create the parent directory structure
+                if (destinationParent.mkdirs()) {
+                    log.debug("Creation of folder is successful. Directory Name : " + destinationParent.getName());
```

Comment:
can we change this to "Successfully created folder " and then append the folder name?